src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -230,9 +240,20 @@ private XElement GetPackageMappingsElementForSource(string packageSource)
                 {
                     pkgSrc.Add(new XElement("package", new XAttribute("pattern", packagePattern)));
                 }
-                else
+                else if (packageSource.Equals(PreviouslySourceBuiltSourceName))
                 {
-                    foreach (string version in previouslyBuiltPackages[packagePattern])
+                    foreach (string version in previouslySourceBuiltPackages[packagePattern])
+                    {
+                        if (!currentPackages[packagePattern].Contains(version))
+                        {
+                            pkgSrc.Add(new XElement("package", new XAttribute("pattern", packagePattern)));
+                            break;
+                        }
+                    }
+                }
+                else // prebuilt source
```

Comment:
Consider asserting this is the prebuilt feed.  I tend to think it would be safer to make this an if check and have the else throw/no-op.  It helps future proof the code better.